File: sdks/go/examples/basic.go
package main

import (
	"context"
	"fmt"
	"log"
	"time"
)

// ThemisDB Go SDK - Basic Usage Example
//
// This example demonstrates basic operations with the ThemisDB Go client.
// Note: This is a placeholder example. Actual implementation pending.

func main() {
	fmt.Println("ThemisDB Go SDK - Basic Example")
	fmt.Println("================================\n")

	// 1. Initialize the client
	fmt.Println("1. Initializing ThemisDB client...")
	config := ClientConfig{
		BaseURL:     "http://localhost:8080",
		BearerToken: "your-jwt-token-here",
		Timeout:     30 * time.Second,
	}
	fmt.Printf("   Connected to: %s\n\n", config.BaseURL)

	ctx := context.Background()

	// 2. Perform a simple query
	fmt.Println("2. Executing AQL query...")
	query := "FOR doc IN myCollection LIMIT 10 RETURN doc"
	fmt.Printf("   Query: %s\n", query)
	fmt.Println("   Result: (Placeholder - will execute when SDK is implemented)\n")

	// 3. LLM inference example
	fmt.Println("3. LLM Inference...")
	inferRequest := InferRequest{
		Prompt:    "What is ThemisDB?",
		Model:     "mistral-7b",
		MaxTokens: 100,
	}
	fmt.Printf("   Prompt: \"%s\"\n", inferRequest.Prompt)
	fmt.Printf("   Model: %s\n", inferRequest.Model)
	fmt.Println("   Response: (Placeholder - will execute when SDK is implemented)\n")

	// 4. Check health status
	fmt.Println("4. Checking server health...")
	fmt.Println("   Status: (Placeholder - will execute when SDK is implemented)\n")

	fmt.Println("Example completed! SDK structure in place, implementation pending.")

	// Demonstrate context usage
	_ = ctx
}

// Placeholder types for demonstration
type ClientConfig struct {
	BaseURL     string
	BearerToken string
	Timeout     time.Duration
}

type InferRequest struct {
	Prompt    string
	Model     string
	MaxTokens int
}

func init() {
	log.SetFlags(log.LstdFlags | log.Lshortfile)
}

File: sdks/javascript/examples/basic.ts
/**
 * ThemisDB JavaScript SDK - Basic Usage Example
 * 
 * This example demonstrates basic operations with the ThemisDB client.
 * Note: This is a placeholder example. Actual implementation pending.
 */

// Placeholder import - will be actual SDK import once implemented
// import { ThemisDBClient } from '@themisdb/client';

async function basicExample() {
  console.log('ThemisDB JavaScript SDK - Basic Example');
  console.log('=========================================\n');

  // 1. Initialize the client
  console.log('1. Initializing ThemisDB client...');
  const config = {
    baseUrl: 'http://localhost:8080',
    bearerToken: 'your-jwt-token-here'
  };
  console.log(`   Connected to: ${config.baseUrl}\n`);

  // 2. Perform a simple query
  console.log('2. Executing AQL query...');
  const query = 'FOR doc IN myCollection LIMIT 10 RETURN doc';
  console.log(`   Query: ${query}`);
  console.log('   Result: (Placeholder - will execute when SDK is implemented)\n');

  // 3. LLM inference example
  console.log('3. LLM Inference...');
  const inferRequest = {
    prompt: 'What is ThemisDB?',
    model: 'mistral-7b',
    max_tokens: 100
  };
  console.log(`   Prompt: "${inferRequest.prompt}"`);
  console.log(`   Model: ${inferRequest.model}`);
  console.log('   Response: (Placeholder - will execute when SDK is implemented)\n');

  // 4. Check health status
  console.log('4. Checking server health...');
  console.log('   Status: (Placeholder - will execute when SDK is implemented)\n');

  console.log('Example completed! SDK structure in place, implementation pending.');
}

// Run the example
if (require.main === module) {
  basicExample().catch(error => {
    console.error('Error running example:', error);
    process.exit(1);
  });
}

export { basicExample };
